katdal/test/test_van_vleck.py

```diff
@@ -0,0 +1,65 @@
+################################################################################
+# Copyright (c) 2020, National Research Foundation (SARAO)
+#
+# Licensed under the BSD 3-Clause License (the "License"); you may not use
+# this file except in compliance with the License. You may obtain a copy
+# of the License at
+#
+#   https://opensource.org/licenses/BSD-3-Clause
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+from __future__ import print_function, division, absolute_import
+from builtins import zip
+
+import numpy as np
+
+from katdal.van_vleck import norm0_cdf, autocorr_lookup_table
+
+
+def test_norm0_cdf():
+    scale = 2.0
+    x = np.array([0.0, 2.0, 4.0, 6.0])
+    # Generated by scipy.stats.norm.cdf(x, scale=2.0)
+    expected = np.array([0.5, 0.8413447460685429, 0.9772498680518208, 0.9986501019683699])
+    actual = norm0_cdf(x, scale)
+    np.testing.assert_allclose(actual, expected, rtol=0., atol=np.finfo(float).eps)
+    actual = norm0_cdf(-x, scale)
+    np.testing.assert_allclose(actual, 1.0 - expected, rtol=0., atol=np.finfo(float).eps)
+    actual = norm0_cdf(x[-1], scale)
+    np.testing.assert_allclose(actual, expected[-1], rtol=0., atol=np.finfo(float).eps)
+
+
+def test_autocorr_correction():
+    # 15-level "4-bit" KAT-7 requantiser
+    levels = np.arange(-7., 8.)
+    quantised_ac_table, true_ac_table = autocorr_lookup_table(levels)
+    N = 100000
+    rs = np.random.RandomState(42)
+    autocorrs = (0.06, 0.2, 1.0, 10.0, 100.0, 1000.0)
+    # Excess above usual sample stdev, generated by Bayesian quantisation correction code
```

Comment:
> Is it to account for the loss of information caused by the quantisation?

Yes.

> And where does one find this "Bayesian quantisation correction code" should one wish to modify the test?

Ask the author and he will provide it 😿 

Ideally this would make it into katdal too at a later stage, but that process does not have a good track record...

Reply:
Documented this.